docs/en/users/10_filter.md

```diff
@@ -115,6 +117,31 @@ Finally, parentheses may be used to express more complex queries, with basic neg
 
 > ℹ️ If you need to search for a parenthesis, it needs to be escaped like `\(` or `\)`
 
+### Regex
+
+Text searches (including `author:`, `intitle:`, `inurl:`, `#`) may use regular expressions, which must be enclosed in `/ /`.
+
+Regex searches are case-sensitive by default, but can be made case-insensitive with the `i` modifier like: `/Alice/i`
+
+Supports multiline mode with `m` modifier like: `/^Alice/m`
+
+> ℹ️ `author:` is working with one author per line, so the multiline mode may advantageously be used, like: `author:/^Alice Dupont$/im`
+>
+> ℹ️ `#` is likewise working with one tag per line, so the multiline mode may advantageously be used, like: `#/^Hello World$/im`
+
+Example to search entries, which title starts with the *Lol* word, with any number of *o*: `intitle:/^Lo+l/i`
+
+As opposed to normal searches, HTML special characters are not escaped in regex searches, to allow searching HTML code, like: `/Hello <span>world<\/span>/`
```

Comment:
This also has the effect that I have to use the HTML encoding of any character literals I'm looking for in the body as well, right? So `&amp;` and `&quot;` to find the post-rendered characters `&` and `"` for example.

Might it be worth doing this only if you have an `r` character on the regex?  Like `/Hello <span>world<\/span>/r`?

Reply:
Indeed, the search is supposed to match the HTML content verbatim. No need to extra encode anything though, just exactly at it appears in the original HTML document.

We could consider adding some modifiers, but let's see if there is a sufficient need first. Another modifier I am considering is an automated transformation from PCRE syntax (subset) to the other variants such as PostgreSQL.

Reply:
I think you might run into issues trying to translate between the regex versions. If you settle on the PCRE syntax there are a lot of advanced `(?...)` features that aren't present in Postgres.  Postgres also has explicit start and end word boundary markers, while PCRE has only a singular word boundary marker.  Then there's MySQL, which supports only a very small subset of regex features.  

I could see maybe trying to do something with the constraint escapes for Postgres, but that's about the only thing I can see that could possibly be translated and it would probably just confuse the situation with the databases having different regex formats.